packages/core/src/types.tsx

```diff
@@ -599,7 +599,7 @@ export type TypedNavigator<
 };
 
 export type NavigatorScreenParams<
-  ParamList,
+  ParamList extends ParamListBase | NavigationState,
```

Comment:
I'm confused why `ParamList` would be of type `NavigationState`

Reply:
@satya164 yeh, it shouldn't be, it could extends ParamListBase only, i will fix that, thanks